### ISSUE TITLE
RFE: Unbound formatted files

### ISSUE THREAD

**Issue by duh-nm** [StevenBlack/hosts#237]
The straight hosts files work on local computers and dnsmasq, but unbound is becoming more popular for small business, and home networks. Unbound formatted files would be nice to have.
I currently download the raw file and run it through this beast to format it, but having it directly in an unbound format would be nice for others as well i would assume.

#!/bin/bash
DATE=$(date +%F)
curl -o /tmp/$DATE https://raw.githubusercontent.com/StevenBlack/hosts/master/alternates/fakenews-gambling-porn/hosts
echo "" > /etc/unbound/local.d/blockservers.conf
for line in $(more $DATE | grep -v "#" |grep -v "localhost" | grep -v "127.0.0.1 local" | grep -v "broadcasthost" | sort | uniq | awk '{print $2}')
do
	echo "local-zone: \"$line\" redirect" >> /etc/unbound/local.d/blockservers.conf
	echo "local-data: \"$line A 0.0.0.0\"" >> /etc/unbound/local.d/blockservers.conf
done

**Comment by StevenBlack** [comment]
Interesting.

Note to self: see also: https://deadc0de.re/articles/unbound-blocking-ads.html

**Comment by StevenBlack** [comment]
@duh-nm  commit f91e14089 adds a link to your work in the readme.

**Comment by ScriptTiger** [comment]
What is the pull request wanted here? For a conversion script to Unbound to be included in the repo here?

**Comment by savyajha** [comment]
There's a repo dedicated to formatting the contents of this repo for dnsmasq and unbound. [Here](https://github.com/oznu/dns-zone-blacklist) it is.

**Comment by ScriptTiger** [comment]
Maybe it can be added to the "Interesting Applications" section. @StevenBlack, is the intention to keep the repo focused on hosts files and provide additional resources in the readme or are you open to including support for additional formats within the repo itself? I personally feel, as I've said in previous issues, getting into other formats would be a bit of a can of worms. With the volume of information being dealt with and the unlimited formats out there, I personally feel as though each format is better suited to have it's own dedicated repo supported elsewhere to get the attention they each deserve.

**Comment by dgoerger** [comment]
If you're running [Unbound 1.5.10+](https://unbound.net/pipermail/unbound-users/2016-September/004466.html), I believe it's slightly more efficient to use `always_nxdomain` (immediately return nxdomain) than `static` (return nxdomain if not found in cache) or `redirect` (respond with an alternate IP, a connection to which then needs to time out). The efficiency gains at normal loads are negligible, sure, but it's still a [however slightly] better code path.

I think this illustrates a good point as to the difficulty of maintaining cross-platform configuration files. I maintain formatting scripts for [unbound](https://github.com/dgoerger/dotfiles/blob/507e6857b95a0026ca663ae060a94c296d5c2f29/scripts/dnsblock.sh) (*nix) and [dnsmasq](https://github.com/dgoerger/zoo_workstation/blob/v2.0.0/files/dnsblock_updater) (Linux-only), but follow Fedora Betas and OpenBSD-CURRENT, i.e., running my confs on a more conservative distro such as Debian may or may not go over very well. At this time, Debian Stable is recently released and has [Unbound 1.6.0](https://packages.debian.org/stretch-updates/unbound), i.e., it's compatible with the post-1.5.10 syntax. [Debian Oldstable](https://packages.debian.org/jessie-updates/unbound) and [Ubuntu Trusty](https://packages.ubuntu.com/trusty-updates/unbound) (supported until April 2019) are not---each featuring Unbound 1.4.22. BIND9.10 introduced great support for a DNS-based firewall-like feature using [Response Policy Zones](https://en.wikipedia.org/wiki/Response_policy_zone#Example_of_use), but on [still-supported BIND9.9](https://packages.ubuntu.com/trusty-updates/bind9) it [might give you trouble, if you try to specify more than one RPZ](https://serverfault.com/a/618119).

I'm not saying it can't be done, just that I was about to file tickets and pull requests to @oznu's excellent [repo](https://github.com/oznu/dns-zone-blacklist) with regard to the newer syntax, and have now decided against it, in the interest of not breaking the DNS for users of e.g. Trusty and Oldstable. In this sense I think it's wiser for users to run a script to format their blocklist---or at the very least, a script which can call `unbound-checkconf` or `named-checkconf`, etc, to verify syntax before trying to load potentially-invalid conf files. Sanitize the input, as they say. 😀

**Comment by Martii** [comment]
For posterity sake had some time to look into why unbound wasn't working with a list from here with jammy:

``` console
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.1 LTS
Release:        22.04
Codename:       jammy

$ apt-cache policy unbound
unbound:
  Installed: 1.13.1-1ubuntu5.1
  Candidate: 1.13.1-1ubuntu5.1
  Version table:
 *** 1.13.1-1ubuntu5.1 500
        500 http://us.archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages
        500 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages
        100 /var/lib/dpkg/status
     1.13.1-1ubuntu5 500
        500 http://us.archive.ubuntu.com/ubuntu jammy/universe amd64 Packages
```

... and getting this error on a restart of unbound:

``` console
Restarting unbound
/etc/unbound/unbound.conf.d/ads.conf:1: error: syntax error
read /etc/unbound/unbound.conf failed: 1 errors in configuration file
[1665961176] unbound-control[34861:0] fatal error: could not read config file
/etc/unbound/unbound.conf.d
```

Re: https://deadc0de.re/articles/unbound-blocking-ads.html *(and other web sources found)*

... the "change" between focal and jammy is that `server:` seems to be needed to be prepended to the separate .conf list. Most of the web docs at this time don't show this. e.g.

```
local-zone: "0.0.0.0" always_nxdomain
...
```

... becomes ...

``` console
server:
local-zone: "0.0.0.0" always_nxdomain
...
```

... that gives this on restart:

``` console
Restarting unbound
ok
/etc/unbound/unbound.conf.d

```

*(note the ellipsis means the rest of the list used and the beginning of the file is that "label")*

`unbound-checkconf` was useless in a separate .conf case and told me everything was okay.

Focal refs:

``` console
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.5 LTS
Release:        20.04
Codename:       focal

$ apt-cache policy unbound
unbound:
  Installed: (none)
  Candidate: 1.9.4-2ubuntu1.3
  Version table:
     1.9.4-2ubuntu1.3 500
        500 http://us.archive.ubuntu.com/ubuntu focal-updates/universe amd64 Packages
        500 http://security.ubuntu.com/ubuntu focal-security/universe amd64 Packages
     1.9.4-2ubuntu1 500
        500 http://us.archive.ubuntu.com/ubuntu focal/universe amd64 Packages
```

Hope this helps someone else out.  Keep up the great work here. THANK YOU! :smile_cat:
